src/admin/api.js

```diff
@@ -349,6 +353,102 @@ function extractDestinationCurrency(currencyPair) {
+                BROKEN: 'BROKEN', // Variable date.
+            };
+
+            /**
+             * rateSetId {String} - Unique identifier for each set of rates. ^[0-9]{1,10}$
+             * currencyPair {String} - Identifies the currency pair in 6 characters.
+             *      The currency codes defined in ISO 4217 as three-letter alphabetic codes are
+             *      used for each member of the currency pair. ^[A-Z]{6}$
+             * baseCurrency {String} - The base currency of the currency pair. ^[A-Z]{3}$
+             * ratePrecision {String} - The number of decimal places to which this currency pair is
+             *      quoted by the partner bank. NOTE: Use this if the base currency is EUR.
+             *      It should be the same as the decimalRate.
+             * invRatePrecision {String} - Used when exchange rates are stored both ways.
+             *      NOTE: Currently not used.
+             * tenor {String} - tenor must be TN
+             * valueDate {String} - The value date for the rate as per the partner bank.
+             *      NOTE: This is currently not mandatory. It can be null or not present.
+             * bidSpotRate {String} The rate where the partner bank buys the base currency at.
+             *      NOTE: Currently, this must be the same as rate but with the decimal point.
+             * offerSpotRate {String} - The rate where the partner bank sells the base currency at.
+             *      NOTE: This is currently not in use.
+             * midPrice {String} - The rate that is at the middle of both the bid and the offer.
+             *      NOTE: This is currently not in use.
+             * validUntilTime {String} - Time and date that this rate will expire.
+             *      Excludes grace period. Format: yyyy-MM-dd HH:mm:ss.SSS in UTC time.
+             *      NOTE: Currently, this must be the same as endTime but in a different format.
+             *      ^\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}(\.\d{3})?$
+             * isValid {String} - Indicator that the partner bank was able to provide the rate.
+             *      NOTE: Currently this must always be 'true' or 'TRUE'.
+             * isTradable {String} - A flag to indicate that the currency pair is tradeable.
+             *      NOTE: Currently this must always be 'true' or 'TRUE'.
+             */
+            return {
+                rateSetId: RATE_ID_PER_CURRENCY_PAIR[currencyPair.toUpperCase()],
+                currencyPair: currencyPair.toUpperCase(),
+                baseCurrency: extractSourceCurrency(currencyPair).toUpperCase(),
+                ratePrecision: rateDetails.decimalRate,
+                invRatePrecision: '1', // This is currently not in use.
+                tenor: TENOR_VALUES.TN, // Since payment settlement happens on T+1,
+                // where T = date of order execution, tenor must be TN.
+                valueDate: null, // This is currently not in use.
```

Comment:
There's code in `finance-portal-ui` to move from ISO strings to the `2020-03-03` format expected here, which may be needed or not. We should probably move the unit tests and code from there to here.

Reply:
I cannot find it, where is it exactly?

Reply:
https://github.com/mojaloop/finance-portal-ui/blob/caa483e861bdde3745495173644b0cb0d74e470f/src/components/DatePicker.js#L7

Reply:
I chose to not to as the he format expected here is like `2020-04-07 10:29:06.856 `, which is the ISO date time `2020-04-07T10:29:06.856Z` withouth the `T` and the `Z`.

Reply:
Not according to an example I've seen. `valueDate` is `2020-03-03`.

Reply:
You are correct about `valueDate` format, but this is `validUntilTime` which has the format I mention. `valueDate` is set to `null` as it is currently not used.